upload/extension/opencart/catalog/model/fraud/fraudlabspro.php

```diff
@@ -0,0 +1,292 @@
+		}
+
+		if (isset($_SESSION['flp_cc_bin'])) {
+			unset($_SESSION['flp_cc_bin']);
+		}
+
+		if (isset($_SESSION['flp_cc_hash'])) {
+			unset($_SESSION['flp_cc_hash']);
+		}
+
+		// Do not perform any action if error found
+		if ($json->fraudlabspro_error_code) {
+			$this->write_debug_log('Error code:' . $json->fraudlabspro_error_code . ' found in Order ' . (int)$data['order_id'] . '.');
+			return;
+		}
+
+		if (($this->config->get('fraud_fraudlabspro_zapier_approve') && $json->fraudlabspro_status == 'APPROVE') || ($this->config->get('fraud_fraudlabspro_zapier_review') && $json->fraudlabspro_status == 'REVIEW') || ( $this->config->get('fraud_fraudlabspro_zapier_reject') && $json->fraudlabspro_status == 'REJECT')) {
+			// Use zaptrigger API to get zap information
+			$zap_request['key'] = $this->config->get('fraud_fraudlabspro_key');
+			$zap_request['format'] = 'json';
+
+			$curl = curl_init();
+			curl_setopt($curl, CURLOPT_URL, 'https://api.fraudlabspro.com/v1/zaptrigger?' . http_build_query($zap_request));
+			curl_setopt($curl, CURLOPT_HEADER, 0);
+			curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, 0);
+			curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+			curl_setopt($curl, CURLOPT_FORBID_REUSE, 1);
+			curl_setopt($curl, CURLOPT_FRESH_CONNECT, 1);
+
+			$zap_response = curl_exec($curl);
+
+			curl_close($curl);
+
+			if (is_null($zap_json = json_decode($zap_response)) === FALSE) {
+				$target_url = $zap_json->target_url;
+			} else {
+				$target_url = '';
+			}
+
+			if (!empty($target_url)) {
+				$zapresponse = $this->http($target_url, [
+					'id'			=> $json->fraudlabspro_id,
+					'date_created'	=> gmdate('Y-m-d H:i:s'),
+					'flp_status'	=> $json->fraudlabspro_status,
+					'full_name'		=> $data['firstname'] . ' ' . $data['lastname'],
+					'email'			=> $data['email'],
+					'order_id'		=> $data['order_id'],
+				]);
+				$zapdata = json_decode($zapresponse);
+				if (is_object($zapdata)) {
+					if ($zapdata->status == 'success') {
+						$this->write_debug_log('Hooks sent to Zapier successful.');
+					} else {
+						$this->write_debug_log('Hooks sent to Zapier failed.');
+					}
+				} else {
+					$this->write_debug_log('Failed in sending hook to Zapier.');
+				}
+			} else {
+				$this->write_debug_log('Zapier target_url not found.');
+			}
+		}
+
+		$this->write_debug_log('FraudLabs Pro validation completed with status [' . $json->fraudlabspro_status . ']. Transaction ID = ' . $json->fraudlabspro_id . '.');
+
+		if ($json->fraudlabspro_status == 'REVIEW') {
+			return $this->config->get('fraud_fraudlabspro_review_status_id');
+		}
+
+		if ($json->fraudlabspro_status == 'APPROVE') {
+			return $this->config->get('fraud_fraudlabspro_approve_status_id');
+		}
+
+		if ($json->fraudlabspro_status == 'REJECT') {
+			return $this->config->get('fraud_fraudlabspro_reject_status_id');
+		}
+	}
+
+	private function hashIt($s) {
+		$hash = 'fraudlabspro_' . $s;
+
+		for ($i = 0; $i < 65536; $i++)
```

Comment:
There used to be a lot of debate regarding that number range. Perhaps pulling the official range number from their API documentation for accurate result would help.